In [2]:
import pandas as pd

# just like import and then find the ones that are special, classify what part is special
# then print
movement = pd.read_csv("movement_checked.csv")
age = pd.read_csv("age_checked.csv")

movement_ids = set(movement["ID"])
movement_flagged = set(movement.loc[movement["movement_flag"], "ID"])

age_usable = set(age.loc[age["result"] != "not enough to compare", "ID"])

age_flagged = set(age.loc[age["result"].isin(["flagged younger", "flagged older"]), "ID"])

both = movement_flagged & age_flagged

print("The IDs flagged in both were: ", sorted(both))

age_only = []
age_no_movement = []

for cluster_id in age_flagged:
    if cluster_id not in movement_ids:
        age_no_movement.append(cluster_id)
    elif cluster_id not in movement_flagged:
        age_only.append(cluster_id)

movement_only = []
movement_no_usable_age = []

for cluster_id in movement_flagged:
    if cluster_id not in age_usable:
        movement_no_usable_age.append(cluster_id)
    elif cluster_id not in age_flagged:
        movement_only.append(cluster_id)

print("Age was flagged while movement measured but not flagged:", sorted(age_only))
print("Age was flagged while movement unavailable:", sorted(age_no_movement))
print("Movement was flagged while age checked but not flagged:", sorted(movement_only))
print("Movement was flagged while no usable age comparison was found:", sorted(movement_no_usable_age))

The IDs flagged in both were:  ['NGC 1851']
Age was flagged while movement measured but not flagged: ['NGC 6171', 'Pal 12']
Age was flagged while movement unavailable: []
Movement was flagged while age checked but not flagged: ['NGC 3201', 'NGC 6101', 'NGC 6205', 'NGC 6584', 'NGC 6681', 'NGC 6934', 'NGC 6981']
Movement was flagged while no usable age comparison was found: ['2MS-GC02', 'Djorg 1', 'NGC 5139', 'NGC 6287', 'NGC 6333', 'NGC 6528', 'NGC 7006']


## Comparing the age and movement results

Match the clusters with the IDs in both the datasets. NGC 1851 was the only cluster that was flagged completely in both the datasets. Its measured movement is among the highest in the HRS sample, and in the Crawford data, its age also stands out. But Vandenberg gives a different age. So the age evidence for this cluster is not settled. 

Pal 12 and NGC 6171 were both flagged for age, but they were not flagged for movement. Pal 12 also has a similar age in both age datasets, while NGC 6171's age changes when we adjust the rule. Not having a movement flag does not rule out either cluster. It just means that the measured speed did not meet my chosen cutoff, the cutoff, whose reason we already explained.

We also had seven movement flagged clusters that had a usable age comparison but was not really flagged for age. Also that we had another seven that didn't even have an age comparison, so I cannot really say that their ages are unusual under my rules. None of the age flagged clusters relate to measurement movement. 

Overall NGC 1851 is the clearest overlap to look into, but these two overlaps do not necessarily establish it being from another galaxy. The other groups are useful too, especially if missing uncertain age prevents a fair comparison.